cmd/gardener-scheduler/app/gardener_scheduler.go

```diff
@@ -111,175 +77,112 @@ func NewCommandStartGardenerScheduler() *cobra.Command {
+	if err := schedulerfeatures.FeatureGate.SetFromMap(config.FeatureGates); err != nil {
+		return fmt.Errorf("failed to set feature gates: %w", err)
 	}
 
-	k8sGardenClient, err := clientMap.GetClient(ctx, keys.ForGarden())
+	// Initialize logger
+	zapLogger, err := logger.NewZapLogger(config.LogLevel)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get garden client: %w", err)
+		return fmt.Errorf("failed to init logger: %w", err)
 	}
 
-	// Set up leader election if enabled and prepare event recorder.
-	var (
-		leaderElectionConfig *leaderelection.LeaderElectionConfig
-		recorder             = utils.CreateRecorder(k8sGardenClient.Kubernetes(), "gardener-scheduler")
-	)
-
-	if cfg.LeaderElection.LeaderElect {
-		k8sGardenClientLeaderElection, err := kubernetesclientset.NewForConfig(restCfg)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create garden client for leader election: %w", err)
+	sugarLogger := zapLogger.Sugar()
+	defer func() {
+		if err := sugarLogger.Sync(); err != nil {
+			fmt.Println(err)
 		}
+	}()
 
-		leaderElectionConfig, err = utils.MakeLeaderElectionConfig(
-			cfg.LeaderElection.LeaderElectionConfiguration,
-			cfg.LeaderElection.LockObjectNamespace,
-			cfg.LeaderElection.LockObjectName,
-			k8sGardenClientLeaderElection,
-			recorder,
-		)
-		if err != nil {
-			return nil, err
-		}
-	}
+	sugarLogger.Info("Starting Gardener scheduler ...")
+	sugarLogger.Infof("Feature Gates: %s", schedulerfeatures.FeatureGate.String())
 
-	return &GardenerScheduler{
-		Config:         cfg,
-		ClientMap:      clientMap,
-		Logger:         logger,
-		Recorder:       recorder,
-		LeaderElection: leaderElectionConfig,
-	}, nil
-}
+	// set the logger used by sigs.k8s.io/controller-runtime
+	zapLogr := logger.NewZapLogr(zapLogger)
+	ctrlruntimelog.SetLogger(zapLogr)
 
-// Run runs the Gardener Scheduler. This should never exit.
-func (g *GardenerScheduler) Run(ctx context.Context) error {
-	controllerCtx, controllerCancel := context.WithCancel(ctx)
-	defer controllerCancel()
+	// Prepare a Kubernetes client object for the Garden cluster which contains all the Clientsets
+	// that can be used to access the Kubernetes API.
+	if kubeconfig := os.Getenv("KUBECONFIG"); kubeconfig != "" {
+		config.ClientConnection.Kubeconfig = kubeconfig
+	}
 
-	// Prepare a reusable run function.
-	run := func(ctx context.Context) {
-		g.startScheduler(ctx)
+	// Prepare REST config
+	restCfg, err := kubernetes.RESTConfigFromClientConnectionConfiguration(&config.ClientConnection, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create Kubernetes REST configuration: %w", err)
+	}
+
+	// Setup controller-runtime manager
+	mgr, err := manager.New(restCfg, manager.Options{
+		MetricsBindAddress:         getAddress(config.MetricsServer),
+		HealthProbeBindAddress:     getHealthAddress(config),
+		LeaderElection:             config.LeaderElection.LeaderElect,
+		LeaderElectionID:           "gardener-scheduler-leader-election",
```

Comment:
```suggestion
		LeaderElectionID:           config.LeaderElection.ResourceName,
```

Arg, I right now figured, that we have two fields in the `LeaderElectionConfiguration` for namespace and name each 😕 
This should be cleaned up.

Reply:
➡️ https://github.com/gardener/gardener/issues/4380

Reply:
So we keep it like this in this PR?

Reply:
See c5ae64c